Doc/library/signal.rst

```diff
@@ -677,3 +680,52 @@ Do not set :const:`SIGPIPE`'s disposition to :const:`SIG_DFL`
 in order to avoid :exc:`BrokenPipeError`.  Doing that would cause
 your program to exit unexpectedly also whenever any socket connection
 is interrupted while your program is still writing to it.
+
+.. _handlers-and-exceptions:
+
+Note on Signal Handlers and Exceptions
+--------------------------------------
+
+If a signal handler raises an exception, the exception will be propagated to
+the main thread and may be raised after any :term:`bytecode` instruction. Most
+notably, a :exc:`KeyboardInterrupt` may appear at any point during execution.
```

Comment:
The example in your issue (KeyboardInterrupt in the middle of an `__enter__` was very helpful for me in understanding the problem. What do you think of adding it here?

Reply:
I added a simple example after this paragraph.